### ISSUE TITLE
Set up a demopage

### ISSUE THREAD

**Issue by johnnychen94** [JeffFessler/mirt-demo#1]
Similar to https://juliaimages.org/latest/democards/examples/, I could help set up a demo page here using [DemoCards](https://github.com/johnnychen94/DemoCards.jl) if you like it.

Advantages:

* demos are just Julia scripts with [Literate](https://github.com/fredrikekre/Literate.jl) syntax
* notebooks, cover images are generated on the fly
* demo layout is managed automatically

Check and see how demos are organized https://github.com/JuliaImages/juliaimages.github.io/tree/source/docs/examples 

Of course, if you want, you could also host demos in the JuliaImages docs repo, people there could help manage and maintain the demos.

**Comment by JeffFessler** [comment]
Wow, those look really useful.
I think it would be preferable to host the demos either at MIRT.jl site itself, or at `JuliaImages`, rather than in this little repo here that I threw together quickly for a tutorial I was giving.
I really appreciate your offer to help so I'll defer to you which location you think makes more sense.
I am currently doing an overhaul of MIRT.jl to isolate the test code from the src so that hopefully it should precompile faster.

**Comment by johnnychen94** [comment]
> I am currently doing an overhaul of MIRT.jl to isolate the test code from the src so that hopefully it should precompile faster.

yes, moving test-only codes to `test/` definitely helps thining the codebase and improves the precompilation.

Moving plots and IO related codes out can also improve the precompilation.

> I think it would be preferable to host the demos either at MIRT.jl site itself, or at JuliaImages, 

I just noticed that there's no docs page for MIRT. In that case, I can help set these up in MIRT.jl. But after that, it would still be your work to filling the docs as I know very little about the MIRT internal.

**Comment by JeffFessler** [comment]
That would be wonderful.  It has been on my "todo" list for a while but I keep working on the code instead of the docs...  Having a doc & demo setup would let me move some (perhaps all) of the plots out of the package and into the docs/demos so that the package itself is lighter.  Thanks!

**Comment by johnnychen94** [comment]
Would you mind checking and updating the demo scripts to make sure it runs without GUI? I'm unfamiliar with the tools you're using and it's a bit challenging for me to adjust the syntax in a way you like.

I tried one script but it seems not working now.

https://github.com/JeffFessler/mirt-demo/blob/360183f617f92dab30f5adf5032486e2bd8bd007/mri/mri-sim-2d%2Bt.jl#L93

```julia
julia> ir_mri_sensemap_sim(dims=N, ncoil=ncoil, orbit_start=90)
ERROR: TypeError: in keyword argument orbit_start, expected AbstractArray{var"#s141",1} where var"#s141"<:Real, got a value of type Int64
Stacktrace:
 [1] ir_mri_sensemap_sim(; kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol,Symbol},NamedTuple{(:dims, :ncoil, :orbit_start),Tuple{Tuple{Int64,Int64},Int64,Int64}}}) at /Users/jc/Documents/Julia/MIRT.jl/src/mri/sensemap-sim.jl:56
 [2] top-level scope at REPL[29]:1
```

**Comment by JeffFessler** [comment]
Oops, this demo needs an update due to some of the breaking changes I've made to MIRT.
I'll get to this soon.

**Comment by JeffFessler** [comment]
@johnnychen94,
I have tagged a new version of MIRT with our many changes and now I have updated the code in mirt-demo.
The .jl demo files run fine for me now with the latest version.
(The binder files take very long to start so I am unsure if they are running.)

**Comment by johnnychen94** [comment]
I'll take a try in the next few days
